File: go/elons-toys/elons_toys.go
package elon

import "fmt"

// Drive updates the number of meters driven based on the car's speed,
// and reduces the battery according to the battery drainage.
func (car *Car) Drive() {
	if car.battery >= car.batteryDrain {
		car.distance += car.speed
		car.battery -= car.batteryDrain
	}
}

// DisplayDistance return the distance as displayed on the LED display.
func (car *Car) DisplayDistance() string {
	return fmt.Sprintf("Driven %d meters", car.distance)
}

// DisplayBattery returns the battery percentage as displayed on the LED display.
func (car *Car) DisplayBattery() string {
	return fmt.Sprintf("Battery at %d%%", car.battery)
}

// CanFinish takes a trackDistance int as its parameter and returns true if the car can finish the race;
// otherwise, return false
func (car *Car) CanFinish(trackDistance int) bool {
	return trackDistance/car.speed <= car.battery/car.batteryDrain
}

File: csharp/elons-toys/ElonsToys.cs
using System;

class RemoteControlCar
{
    private int _distanceDriven = 0;
    private int _batteryCharge = 100;
    public static RemoteControlCar Buy()
    {
        return new RemoteControlCar();
    }

    public string DistanceDisplay()
    {
        return string.Format($"Driven {_distanceDriven} meters");
    }

    public string BatteryDisplay()
    {
        if (_batteryCharge > 0)
        {
            return string.Format($"Battery at {_batteryCharge}%");
        }
        else
        {
            return string.Format($"Battery empty");
        }
    }

    public void Drive()
    {
        if (_batteryCharge > 0)
        {
            _distanceDriven += 20;
            _batteryCharge -= 1;
        }
    }
}
